pages/scan-results/ScanResults.tsx

```diff
@@ -179,7 +229,16 @@ class ScanResults extends Component {
     })
   }
 
-  updatePackageState(pack, state) {
+  updatePackageState(
+    pack: Package,
+    state:
+      | {
+          promiseState: 'fulfilled'
+          result: Result
+          version?: string | undefined
+        }
+      | { promiseState: 'rejected'; result: undefined; error: any }
```

Comment:
Let's move this type definition outside this function definition for clarity

Reply:
Done